js/screens/authentication/SpidInformationRequestScreen.js

```diff
@@ -0,0 +1,131 @@
+// @flow
+
+import * as React from 'react'
+import { connect } from 'react-redux'
+import { isValid } from 'redux-form'
+import {
+  type NavigationScreenProp,
+  type NavigationState
+} from 'react-navigation'
+import {
+  Container,
+  Content,
+  Text,
+  View,
+  Button,
+  Icon,
+  Header,
+  Left,
+  Body,
+  H1
+} from 'native-base'
+
+import { type GlobalState } from '../../reducers/types'
+import I18n from '../../i18n'
+import Modal from '../../components/ui/Modal'
+import SpidInformationForm, {
+  FORM_NAME
+} from '../../components/forms/SpidInformationForm'
+
+type ReduxMappedProps = {
+  isFormValid: boolean
+}
+
+type OwnProps = {
+  navigation: NavigationScreenProp<NavigationState>
+}
+
+type Props = ReduxMappedProps & OwnProps
+
+type State = {
```

Comment:
missing `isFormValid` from `State` type

Reply:
isFormValid is not a Component state property. Is it a ReduxMappedProps

````
type ReduxMappedProps = {
  isFormValid: boolean
}
```

and we are using the `isValid` redux-form state selector to populate it:

```
const mapStateToProps = (state: GlobalState): ReduxMappedProps => ({
  /**
   * Our form submit button is outside the `Form` itself so we need to use
   * this selector to check if the form is valid or not.
   */
  isFormValid: isValid(FORM_NAME)(state)
})
```